src/main/app/pdf/mappers/interestMapper.ts

```diff
@@ -18,17 +19,24 @@ export class InterestMapper {
       return {
         rate: claimData.interest.rate,
         dateClaimedFrom: MomentFormatter.formatLongDate(interestDate),
-        claimedAtDateOfSubmission: NumberFormatter.formatMoney(InterestMapper.calculateInterest(claimData)),
+        claimedAtDateOfSubmission: NumberFormatter.formatMoney(InterestMapper.calculateInterest(claimData, interestDate)),
         accruedInterest: NumberFormatter.formatMoney(InterestMapper.calculateDailyAmount(claimData))
       }
     }
   }
 
-  public static calculateInterest (claimData: ClaimData): number {
-    return calculateInterest(claimData.amount, claimData.interest, claimData.interestDate.date)
+  public static calculateInterest (claimData: ClaimData, interestDate: Moment): number {
+    return calculateInterest(this.totalAmount(claimData), claimData.interest, interestDate)
   }
 
   private static calculateDailyAmount (claimData: ClaimData): number {
-    return claimData.amount / claimData.interest.rate / 365
+    return this.totalAmount(claimData) / claimData.interest.rate / 365
+  }
+
+  private static totalAmount (claimData: ClaimData): number {
+    if (claimData.amount) {
+      return claimData.amount.totalAmount()
+    }
+    return 0.00
```

Comment:
Application should file if amount at PDF generation time is not specified. This just hides possible errors.

Change needed.

Reply:
Done